syntax = "proto3";

package category;

service CategoryService {
    // Create a new category
    rpc CreateCategory(CreateCategoryRequest) returns (CategoryResponse);

    // Get a category by its UUID
    rpc GetCategory(GetCategoryRequest) returns (CategoryResponse);

    // Update a category
    rpc UpdateCategory(UpdateCategoryRequest) returns (CategoryResponse);

    // Delete a category by its UUID
    rpc DeleteCategory(DeleteCategoryRequest) returns (DeleteCategoryResponse);

    // List all categories by store
    rpc ListCategories(ListCategoriesRequest) returns (ListCategoriesResponse);
}

// Request and response messages
message CreateCategoryRequest {
    string  StoreUuid = 1;
    string  Name = 2;
    string  Description = 3;
    optional string ParentCategoryUuid = 4; // optional parent category
}

message GetCategoryRequest {
    string CategoryUuid = 1;
}

message UpdateCategoryRequest {
    string CategoryUuid = 1;
    optional string Name = 2;
    optional string Description = 3;
    optional string ParentCategoryUuid = 4;
}

message DeleteCategoryRequest {
    string CategoryUuid = 1;
}

message ListCategoriesRequest {
    string StoreUuid = 1;
}

message CategoryResponse {
    bool Success = 1;
    string CategoryUuid = 2;
    string Name = 3;
    string Description = 4;
    string ParentCategoryUuid = 5;
}

message ListCategoriesResponse 
{
    repeated CategoryResponse categories = 1;
}

message DeleteCategoryResponse 
{
    bool Success = 1;
}
